core/services/gateway/connector/connector.go

```diff
@@ -71,24 +72,27 @@ func (c *gatewayConnector) Name() string { return c.lggr.Name() }
 
 type gatewayState struct {
 	conn     network.WSConnectionWrapper
-	config   ConnectorGatewayConfig
+	config   config.ConnectorGatewayConfig
 	url      *url.URL
 	wsClient network.WebSocketClient
 }
 
-func NewGatewayConnector(config *ConnectorConfig, signer Signer, handler GatewayConnectorHandler, clock clockwork.Clock, lggr logger.Logger) (GatewayConnector, error) {
-	if config == nil || signer == nil || handler == nil || clock == nil || lggr == nil {
+func NewGatewayConnector(configLocal *config.ConnectorConfig, signer Signer, handler GatewayConnectorHandler, clock clockwork.Clock, lggr logger.Logger) (GatewayConnector, error) {
+	if configLocal == nil || signer == nil || handler == nil || clock == nil || lggr == nil {
 		return nil, errors.New("nil dependency")
 	}
-	if len(config.DonId) == 0 || len(config.DonId) > network.HandshakeDonIdLen {
+	// configLocal.DonId undefined
+	// (type *"github.com/smartcontractkit/chainlink/v2/core/config".ConnectorConfig
+	// is pointer to interface, not interface)compilerMissingFieldOrMethod
+	if len(configLocal.DonId) == 0 || len(configLocal.DonId) > network.HandshakeDonIdLen {
```

Comment:
Additionally, interfaces define a collection of methods a type must have; so the only way to access the underlying type is to call the methods. In other words, you have to call methods to access any attributes.

configLocal.DonId -> configLocal.DonID